example/main.go

```diff
@@ -12,41 +11,40 @@ import (
 
 func main() {
 	vecty.RenderBody(&body{
-		store: vstore.CreateStore(&store.State{}),
+		store: vstore.New(&store.State{}),
 	})
 }
 
 type body struct {
 	vecty.Core
-	store *vstore.Store
+	store vstore.Store
 }
 
 func (b *body) Render() *vecty.HTML {
 	return elem.Body(
-		b.store.NewComponent(NewMainComp),
+		b.store.Connect(NewMainComp()),
 	)
 }
 
 type mainComp struct {
 	vecty.Core
-	Number   int `vecty:"prop"`
-	Dispatch func(action interface{})
+	store vstore.Store
 }
 
 // NewMainComp is the mainComp constructor
-func NewMainComp(s *vstore.Store) vecty.Component {
-	state := s.GetState().(*store.State)
+func NewMainComp() vstore.StoreComponent {
+	return &mainComp{}
+}
 
-	return &mainComp{
-		Number:   state.Number,
-		Dispatch: s.Dispatch,
-	}
+func (comp *mainComp) Connect(store vstore.Store) {
+	comp.store = store
```

Comment:
here we can just do 

```go
state := m.store.State().(*store.State)

comp.Dispatch = store.Dispatch
comp.Number = state.Number
```

Reply:
Yes, either way works, we should continue to provide both options.

Reply:
@norunners i tested this and it didn't work unless you call `connect` from the `dispatch` function. Can you double check? Thanks!

Reply:
I believe we should continue to call connect before rerender. Although it would work if we have the state reference as a component struct field instead of the number int